### PR TITLE
refactor(all): refactor to use data-testid attribute

### PR DESCRIPTION
**Pull request by petermakowski**
## Done
- replace all occurences of `data-test` with `data-testid` to follow our convention
- add eslint rule for data-test attribute to prevent its use in the future
image
- update `react-components` to 0.24.0

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

## Fixes

Fixes: # .

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against *master* rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in master.

### CODE REVIEW

**Comment by Caleb-Ellis** [comment]
Adding a "Don't merge" label for now until 3.1 proper is released, in order to avoid conflicts in case we need to backport any fixes from master. In the meantime maybe we could get `react-components` updated and have this PR handle it all at once.

**Comment by petermakowski** [comment]
> Adding a "Don't merge" label for now until 3.1 proper is released, in order to avoid conflicts in case we need to backport any fixes from master. In the meantime maybe we could get `react-components` updated and have this PR handle it all at once.

Updated in `react-components` here: https://github.com/canonical-web-and-design/react-components/pull/642

**Comment by petermakowski** [comment]
> 

`data-testid` update has been released https://github.com/canonical-web-and-design/react-components/releases/tag/v0.24.0

**Comment by Caleb-Ellis** [comment]
Removing "Don't merge" now that [3.1 is officially out](https://launchpad.net/maas/3.1/3.1.0).

File: ui/src/setupTests.ts [hunk 1]
```diff
@@ -1,13 +1,11 @@
-import { configure } from "@testing-library/react";
+import "@testing-library/react";
 import "@testing-library/jest-dom";
 import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
 import Enzyme from "enzyme";
 import { enableFetchMocks } from "jest-fetch-mock";
 
 Enzyme.configure({ adapter: new Adapter() });
 
-configure({ testIdAttribute: "data-test" });
```
Comment (hatched) [inline ui/src/setupTests.ts:9]: 👍

File: eslint/react.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+module.exports = {
```
Comment (hatched) [inline eslint/react.js:1]: Thanks for adding this check 👍

**Review by hatched — APPROVED** [review]

**Pull request merged**